### PR TITLE
Refactor accel calibrator to use new math and be non blocking & support gimbal accel cal

### PR DESCRIPTION
**Pull request by jschall**
This replaces https://github.com/diydrones/ardupilot/pull/3058 - the only difference as of now is that this one has been rebased.

I'll take over maintenance of the PR.

### CODE REVIEW

**Comment by jschall** [comment]
@bugobliterator did you resolve Tridge's concerns in the old PR? You committed changes but didn't leave a comment.

**Comment by jschall** [comment]
@tridge This breaks all examples that include AP_InertialSensor. The options are to move AP_AccelCal into AP_InertialSensor or to add
    LIBRARIES += AP_AccelCal
to almost every single make.inc in libraries.

What's your preference?

**Comment by tridge** [comment]
@jschall I'd be happy with either approach

**Comment by jschall** [comment]
@tridge agreed.

Want to take a peek at the Travis build log and help me figure out why it is failing?

**Comment by jschall** [comment]
Closed pending changes

File: Tools/ardupilotwaf/ardupilotwaf.py [hunk 1]
```diff
@@ -28,6 +28,7 @@
     'AP_GPS',
     'AP_HAL',
     'AP_HAL_Empty',
+    'AP_AccelCal',
     'AP_InertialSensor',
     'AP_Math',
     'AP_Mission',
```
Comment (lucasdemarchi) [inline Tools/ardupilotwaf/ardupilotwaf.py]: This list is maintained alphabetically sorted to ease maintenance. I'm pushing a fix.

**Pull request closed without merging**